gamla/functional_generic.py

```diff
@@ -474,3 +474,23 @@ def countby_many(f):
             lambda x, y: x + 1 if x else 1,
         ),
     )
+
+
+def merge(*dicts):
+    if len(dicts) == 1 and not isinstance(dicts[0], Mapping):
+        dicts = dicts[0]
+    result_dict = {}
+    for d in dicts:
+        result_dict.update(d)
+    return result_dict
+
+
+def merge_with(func, *dicts):
```

Comment:
this should be curried using a nested functions, so it can be used in pipelines

Reply:
then you can also support an async `f` (since you put it in the async/sync generic file)

Reply:
Yes, so just adding `@currying.curry` ?

Reply:
no, using a nested function